webapp/src/snippetBuilder.tsx

```diff
@@ -25,15 +26,19 @@ interface AnswersMap {
     [answerToken: string]: pxt.SnippetAnswerTypes;
 }
 
-export interface SnippetBuilderState {
+interface SnippetBuilderState {
     visible?: boolean;
     tsOutput?: string;
+    mdOutput?: string;
     answers?: AnswersMap;
     history: number[];
     defaults: DefaultAnswersMap; // Will be typed once more clearly defined
     config?: pxt.SnippetConfig; // Will be a config type
+    actions?: sui.ModalButton[];
 }
 
+let thisBlocksInfo: pxtc.BlocksInfo;
```

Comment:
Leaking state? Is this needed?

Reply:
This I'll need to make a todo for. I'd like to find a better way to provide this to the spriteEditor component because right now its funneled down from blocks.tsx -> snippetBuilder.tsx -> inputHandler.tsx -> spriteEditor.tsx. Maybe that's the only way however it'd be nice if blocks info was available elsewhere in pxt.

Reply:
Ask the worker for the blocksinfo directly.